### PR TITLE
doc: Add guide for writing a plugin in Python

### PR DESCRIPTION
**Pull request by hermanschaaf**
Also updates some other related docs

### CODE REVIEW

File: website/pages/docs/developers/creating-new-plugin/python-source.mdx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+Incremental tables are great for efficiency, but add some additional complexity both on you and on your users. As the plugin author, you should consider first whether the table needs to be incremental, then whether it can be made to be incremental. You can also [read more about managing incremental tables](/docs/advanced-topics/managing-incremental-tables).
+
+## Creating Your First Plugin
+
+As the Python SDK is still new, we don't yet have a plugin scaffold generator. For now, we recommend starting by copying the code from an existing Python plugin. Two reference plugins are [Typeform](https://github.com/cloudquery/cloudquery/tree/main/plugins/source/typeform) and [Square](https://github.com/cloudquery/cloudquery/tree/main/plugins/source/square). The Typeform plugin is relatively simple, and the Square plugin provides an example of generating tables from an OpenAPI spec.
+
+Before running the plugin locally, you will need to install its dependencies, preferably in a virtual environment:
```
Comment (hydratim) [inline website/pages/docs/developers/creating-new-plugin/python-source.mdx:54]: Any particular reason that we aren't using `poetry`?

**Review by yevgenypats — APPROVED** [review]
Looks great. Let's :shipit: !

File: website/pages/docs/developers/creating-new-plugin/python-source.mdx (hunk 1 above)
Comment (hermanschaaf) [reply website/pages/docs/developers/creating-new-plugin/python-source.mdx:54]: Nope, I've just never used it. Is it part of the standard Python toolkit now?

Comment (hydratim) [reply website/pages/docs/developers/creating-new-plugin/python-source.mdx:54]: No, but it's pretty common as it's especially excellent for ensuring your dependencies are up-to-date, and locked to trusted (and hashed) versions.

**Pull request merged**